service/src/main/java/org/mskcc/smile/service/impl/SampleServiceImpl.java

```diff
@@ -81,14 +81,29 @@ public SmileSample saveSmileSample(SmileSample
             if (!existingSample.getSampleClass().equals(sampleMetadata.getTumorOrNormal())) {
                 existingSample.setSampleClass(sampleMetadata.getTumorOrNormal());
             }
-            // determine where a patient swap is required also
+
+            // determine whether a patient swap is required also
             if (!sample.getPatient().getSmilePatientId().equals(
                     existingSample.getPatient().getSmilePatientId())) {
                 LOG.info("Updating sample-to-patient relationship and removing connection to patient: "
                         + existingSample.getPatient().getSmilePatientId());
                 sampleRepository.removeSamplePatientRelationship(existingSample.getSmileSampleId(),
                         existingSample.getPatient().getSmilePatientId());
-                existingSample.setPatient(sample.getPatient());
+                // merge aliases from existing patient to the patient we are swapping to
+                SmilePatient existingPatient = existingSample.getPatient();
```

Comment:
Should we remove existing patient from graph after this?

Reply:
If you were to toggle the smile patient id here I think it would be the same for both of these records. If they do match then that would confirm my suspicion that the table itself is fetching and displaying duplicate results (one record for each sample) instead of results that are actually unique by patient

Reply:
If the smile patient ids for these records to do not match then yeah, that would be indicative of a database issue instead